app/controllers/entries_controller.rb

```diff
@@ -5,8 +5,8 @@ class EntriesController < ApplicationController
 
   def index
     @register = Register.find_by_slug!(params[:register_id])
-    entries = recover_entries_history(@register.id, @register.fields, params)
-    fields = @register.register_fields.map { |field| field['field'] }
+    fields = @register.fields_array
```

Comment:
I don't think we need to create this as variable. Might read easier in the method to just specify the array directly`@register.fields_array`

Reply:
since we are using this multiple times, I assume this ensures we are only doing the DB lookup once, but I am not sure how this works in Ruby/Rails internals.

Reply:
mmm, I only see one call to `#fields_array` in this method. I would expect this field to be fetched from the database every time you instantiate `@register` rather than when you call this method. Am I missing something?

Reply:
Yes, I think you are right, I will amend as-per @danmitchell- suggestion